src/number_format.js

```diff
@@ -795,10 +795,10 @@ class NumberFormat extends React.Component {
     e.persist()
     setTimeout(() => {
       const el = e.target;
-      const {selectionStart, value = ''} = el;
+      const {selectionStart, selectionEnd, value = ''} = el;
 
       const caretPosition = this.correctCaretPosition(value, selectionStart);
-      if (caretPosition !== selectionStart) {
+      if (caretPosition !== selectionStart && !(selectionStart === 0 && selectionEnd === value.length)) {
```

Comment:
Can you add a comment on code for `selectionStart === 0 && selectionEnd === value.length` condition.